docs/docs/the-component-lifecycle.md

```diff
@@ -0,0 +1,43 @@
+---
+id: the-component-lifecycle
+title: The Component Lifecycle
+permalink: docs/the-component-lifecycle.html
+prev: rendering-elements.html
+---
+
+Components have three main parts of their lifecycle:
+
+* **[Mounting](#mounting):** A component is being inserted into the DOM.
+* **[Updating](#updating):** A component is being re-rendered to determine if the DOM should be updated.
+* **[Unmounting](#unmounting):** A component is being removed from the DOM.
+
+React provides lifecycle methods that you can specify to hook into this process. Methods prefixed with **`will`** are called right before something happens, and methods prefixed with **`did`** are called right after something happens.
+
+## Mounting
+
+These methods are called when a component is being inserted into the DOM.
+
+* **`getInitialState(): object`** is invoked before a component is mounted. Stateful components should implement this and return the initial state data.
```

Comment:
@gaearon thanks for your link, which helped me to convert my code to ES6 classes and correct all `getInitialState()` blocks I had because the use of `React.createClass()`